src/avitab/apps/components/ContainerWithClickableCustomList.h

```diff
@@ -0,0 +1,63 @@
+/*
+ *   AviTab - Aviator's Virtual Tablet
+ *   Copyright (C) 2018 Folke Will <folko@solhost.org>
+ *   Copyright (C) 2023 Vangelis Tasoulas <cyberang3l@gmail.com>
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU Affero General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU Affero General Public License for more details.
+ *
+ *   You should have received a copy of the GNU Affero General Public License
+ *   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#ifndef SRC_AVITAB_APPS_COMPONENTS_CONTAINERWITHCLICKABLECUSTOMLIST_H_
+#define SRC_AVITAB_APPS_COMPONENTS_CONTAINERWITHCLICKABLECUSTOMLIST_H_
+
+#include "src/avitab/apps/App.h"
+#include "src/gui_toolkit/widgets/Container.h"
+#include "src/gui_toolkit/widgets/List.h"
+#include "src/gui_toolkit/widgets/Window.h"
+#include <string>
+#include <vector>
+
+namespace avitab {
+
+// dialog that lists a custom set of objects
+class ContainerWithClickableCustomList {
+  public:
+    using CancelCallback = std::function<void(void)>;
+    using SelectCallback = std::function<void(int)>;
+
+    ContainerWithClickableCustomList(App::FuncsPtr appFunctions,
+                                     const std::string &windowTitle);
+
+    void setCancelCallback(CancelCallback cb);
+    void setSelectCallback(SelectCallback cb);
+    void setListItems(const std::vector<std::string> &items);
+    void show(std::shared_ptr<Container> parent);
+    std::string getEntry(uint32_t index);
```

Comment:
I see mixed types for the index in the code, some places uses `int` and others use `uint32_t`. In my opinion, the types should be harmonized and use either `int` (since that's the practical type) or `size_t` (since that's the theoretical limit).

Reply:
I'll convert everything to int as I mostly see ints in the code.

Reply:
I see the problem here and why I used the uint32.

I cannot compare .size (unsigned int) with index (signed int).

I'll `return currentEntries.at(index)`, and this throws an std::out_of_range if the index does not exist.

Reply:
`size_t` would also be fine. Just `uint32_t` seemed a bit too random because nothing is really forcing this to be 32 bits.